obspy/io/csv/core.py

```diff
@@ -0,0 +1,484 @@
+
+        from obspy import read_events
+        names = ('year mon day hour minu sec _ '
+                 'lat lon dep _ _ mag _ _ _ _ _ _ id')
+        catalog = read_events('external.csv', 'CSV', skipheader=1, names=names)
+    """
+    if default is None:
+        default = DEFAULT
+    events = []
+    with _open(fname) as f:
+        for _ in range(skipheader):
+            f.readline()
+        if names is not None:
+            kwargs.setdefault('fieldnames', _names_sequence(names))
+        reader = csv.DictReader(f, **kwargs)
+        for row in reader:
+            if 'time' in row:
+                time = UTC(row['time'])
+            else:
+                time = UTC(
+                    '{year}-{mon}-{day} {hour}:{minu}:{sec}'.format(**row))
+            try:
+                if 'depm' in row:
+                    dep = float(row['depm'])
+                else:
+                    dep = float(row['dep']) * 1000
+                if math.isnan(dep):
+                    raise
+            except Exception:
+                dep = None
+            author = _string(row, 'author')
+            contrib = _string(row, 'contrib')
+            if author is not None or contrib is not None:
+                info = evmod.CreationInfo(author=author, agency_id=contrib)
+            else:
+                info = None
+            origin = evmod.Origin(
+                time=time,
+                latitude=row['lat'],
+                longitude=row['lon'],
+                depth=dep,
+                creation_info=info
+            )
+            try:
+                # add zero to eliminate negative zeros in magnitudes
+                mag = float(row['mag']) + 0
+                if math.isnan(mag):
+                    raise
+            except Exception:
+                magnitudes = []
+            else:
+                try:
+                    magtype = row['magtype']
+                    if magtype.lower() in ('', 'none', 'null', 'nan'):
+                        raise
+                except Exception:
+                    magtype = default.get('magtype')
+                magauthor = _string(row, 'magauthor')
+                info = (evmod.CreationInfo(author=magauthor) if magauthor
+                        else None)
+                magnitudes = [evmod.Magnitude(
+                    mag=mag, magnitude_type=magtype, creation_info=info)]
+            try:
+                id_ = evmod.ResourceIdentifier(row['id'].strip())
+            except Exception:
+                id_ = None
+            region = _string(row, 'region')
+            descs = ([evmod.EventDescription(region, 'region name')]
+                     if region else [])
+            event = evmod.Event(
+                magnitudes=magnitudes,
+                origins=[origin],
+                resource_id=id_,
+                event_descriptions=descs
+            )
+            events.append(event)
+            if format_check:
+                return True
```

Comment:
Certainly optional, but this is a bit long. Maybe can be factored into separate helper functions for reading each part?

Reply:
Let's postpone that